core/src/main/scala/akka/persistence/cassandra/journal/TimeBucket.scala

```diff
@@ -57,4 +57,16 @@ import com.datastax.driver.core.utils.UUIDs
   }
 
   override def toString = s"TimeBucket($key, $startTimestamp, $inPast, $isCurrent)"
+
+  override def equals(other: Any): Boolean = other match {
+    case that: TimeBucket =>
+      key == that.key &&
+        startTimestamp == that.startTimestamp
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(key, startTimestamp)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
```

Comment:
if this is used in hot path I think it should be optimized to avoid the allocations, i.e. using `var` (perhaps with akka.util.HashCode`).

Reply:
Just the equals is, but can't hurt